src/env/environment.rs

```diff
@@ -618,21 +676,13 @@
         );
 
         if default_paths {
-            let mut scstr = paths.data.clone();
-            scstr.push("functions");
-            vars.set_one(
-                L!("fish_function_path"),
-                EnvMode::GLOBAL,
-                str2wcstring(scstr.as_os_str().as_bytes()),
-            );
+            // TODO: Trailing slash?
+            let mut scstr = datadir;
+            scstr.push_str("/functions");
```

Comment:
Probably we control the datadir so there should already be no trailing slash here.
If not, we should "sanitize" the datadir ASAP wherever it is passed in by the user

Reply:
This comes from pathbuf.join everywhere so it's fine